compositor_common/src/scene/builtin_transformations.rs

```diff
@@ -110,16 +114,48 @@ impl FromStr for MirrorMode {
     }
 }
 
+lazy_static! {
+    static ref TRANSFORM_TO_RESOLUTION_CONSTRAINTS: NodeConstraints =
+        NodeConstraints(vec![Constraint::InputsCount(
+            InputsCountConstraint::Exact { fixed_count: 1 }
+        )]);
+    static ref FIXED_POSITION_LAYOUT_CONSTRAINTS: NodeConstraints =
+        NodeConstraints(vec![Constraint::InputsCount(
+            InputsCountConstraint::Range {
+                lower_bound: 1,
+                upper_bound: FIXED_POSITION_LAYOUT_MAX_INPUTS_COUNT,
+            }
+        )]);
+    static ref TILED_LAYOUT_CONSTRAINTS: NodeConstraints =
+        NodeConstraints(vec![Constraint::InputsCount(
+            InputsCountConstraint::Range {
+                lower_bound: 1,
+                upper_bound: FIXED_POSITION_LAYOUT_MAX_INPUTS_COUNT,
+            }
+        )]);
+    static ref MIRROR_IMAGE_CONSTRAINTS: NodeConstraints =
+        NodeConstraints(vec![Constraint::InputsCount(
+            InputsCountConstraint::Exact { fixed_count: 1 }
+        )]);
+    static ref CORNERS_ROUNDING_CONSTRAINTS: NodeConstraints =
+        NodeConstraints(vec![Constraint::InputsCount(
+            InputsCountConstraint::Exact { fixed_count: 1 }
+        )]);
+}
```

Comment:
Can't use normal static, due to heap memory allocation required for `vec`. `[Constraints]` won't work, because `Constraints` isn't sized. `Box` also would allocate on `heap`, requiring runtime to instantiate.